modules/util/timer.go

```diff
@@ -18,3 +19,29 @@ func StopTimer(t *time.Timer) bool {
 	}
 	return stopped
 }
+
+func Debounce(d time.Duration) func(f func()) {
+	type debouncer struct {
+		mu sync.Mutex
+		t  *time.Timer
+	}
+	db := &debouncer{}
+
+	return func(f func()) {
+		db.mu.Lock()
+		defer db.mu.Unlock()
+
+		if db.t != nil {
+			db.t.Stop()
+		}
+		var trigger *time.Timer
+		trigger = time.AfterFunc(d, func() {
+			db.mu.Lock()
+			defer db.mu.Unlock()
+			if trigger == db.t {
+				f()
+			}
```

Comment:
How about
```suggestion
			}
			db.t = nil
```

to release stopped trigger?

Reply:
You can't do it here, otherwise it resets other trigger (eg: trigger != db.t)

But we can put it in the `if trigger == db.t`